internal/cmd/ske/kubeconfig/login/login.go

```diff
@@ -0,0 +1,240 @@
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	clientauthenticationv1 "k8s.io/client-go/pkg/apis/clientauthentication/v1"
+	"k8s.io/client-go/tools/auth/exec"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	expirationSeconds     = 30 * 60          // 30 min
+	refreshBeforeDuration = 15 * time.Minute // 15 min
+)
+
+func NewCmd(p *print.Printer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "login",
+		Short: "Login plugin for kubernetes clients",
+		Long: fmt.Sprintf("%s\n%s",
+			"Login plugin for kubernetes clients, that creates short-lived credentials to authenticate against a STACKIT Kubernetes Engine (SKE) cluster.",
+			"To obtain a kubeconfig for use with the login command, run 'kubeconfig create --login'.",
+		),
+		Args: args.NoArgs,
+		Example: examples.Build(
+			examples.NewExample(
+				`Get a login kubeconfig for the SKE cluster with name "my-cluster". `+
+					"This kubeconfig does not contain any credentials and instead obtains valid credentials via the STACKIT CLI.",
+				"$ stackit ske kubeconfig create my-cluster --login"),
+			examples.NewExample(
+				"Use the previously saved kubeconfig to authenticate to the SKE cluster, in this case with kubectl.",
+				"$ kubectl cluster-info",
+				"$ kubectl get pods"),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			model, err := parseInput()
+			if err != nil {
+				return fmt.Errorf("parseInput: %w", err)
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(p)
+			if err != nil {
+				return err
+			}
+
+			cachedKubeconfig := getCachedKubeConfig(model.CacheKey)
+
+			if cachedKubeconfig == nil {
+				return GetAndOutputKubeconfig(ctx, cmd, apiClient, model, false, nil)
+			}
+
+			certPem, _ := pem.Decode(cachedKubeconfig.CertData)
+			if certPem == nil {
+				_ = cache.DeleteObject(model.CacheKey)
+				return GetAndOutputKubeconfig(ctx, cmd, apiClient, model, false, nil)
+			}
+
+			certificate, err := x509.ParseCertificate(certPem.Bytes)
+			if err != nil {
+				_ = cache.DeleteObject(model.CacheKey)
+				return GetAndOutputKubeconfig(ctx, cmd, apiClient, model, false, nil)
+			}
+
+			// cert is expired, request new
+			if time.Now().After(certificate.NotAfter.UTC()) {
+				_ = cache.DeleteObject(model.CacheKey)
+				return GetAndOutputKubeconfig(ctx, cmd, apiClient, model, false, nil)
+			}
+			// cert expires within the next 15min, refresh (try to get a new, use cache on failure)
+			if time.Now().Add(refreshBeforeDuration).After(certificate.NotAfter.UTC()) {
+				return GetAndOutputKubeconfig(ctx, cmd, apiClient, model, true, cachedKubeconfig)
+			}
+
+			// cert not expired, nor will it expire in the next 15min; therefore, use the cached kubeconfig
+			if err := output(cmd, model.CacheKey, cachedKubeconfig); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	return cmd
+}
+
+type inputModel struct {
```

Comment:
I think this `inputModel` is not needed and it doesnt match the ones that we have for the other commands as this one is not storing any user input. So we can just have the `SKEClusterConfig` struct with the `CacheKey` also defined there